hooks/useThemeSettings/useThemeSettings.ts

```diff
@@ -0,0 +1,37 @@
+import { useNewsroomContext } from '@prezly/theme-kit-nextjs';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
+import { useSessionStorage } from 'react-use';
+
+import type { ThemeSettings } from 'types';
+
+import { DEFAULT_THEME_SETTINGS, STORAGE_KEY } from './constants';
+import { parseQuery } from './utils';
+
+export function useThemeSettings(): ThemeSettings {
+    const { query } = useRouter();
+    const { themePreset } = useNewsroomContext();
+    const [previewSettings, setPreviewSettings] = useSessionStorage(STORAGE_KEY, {});
+
+    const settings: ThemeSettings = {
+        ...DEFAULT_THEME_SETTINGS,
+        ...themePreset?.settings,
+        ...previewSettings,
+    };
+
+    useEffect(() => {
+        const parsedQuery = parseQuery(query);
+        if (Object.keys(parsedQuery).length > 0) {
+            setPreviewSettings(parsedQuery);
+        }
+    }, [query, setPreviewSettings]);
+
+    return {
+        accent_color: settings.accent_color,
+        font: settings.font,
+        header_background_color: settings.header_background_color,
+        header_link_color: settings.header_link_color,
+        show_date: settings.show_date,
+        show_subtitle: settings.show_subtitle,
```

Comment:
I thought about using camel case but went with snake case because I'm using the same interface as the API's `themePreset.settings`.

This way we don't need to maintain and sync two separate interfaces and if there's a property added to `ThemeSettings` interface, it will throw an error if it's missing here and vice versa if it's removed.